public/app/plugins/datasource/grafana-azure-monitor-datasource/azure_log_analytics/response_parser.ts

```diff
@@ -147,6 +147,27 @@ export default class ResponseParser {
   static dateTimeToEpoch(dateTimeValue: any) {
     return dateTime(dateTimeValue).valueOf();
   }
+
+  static parseSubscriptions(result: any): Array<{ text: string; value: string }> {
+    const list: Array<{ text: string; value: string }> = [];
+
+    if (!result) {
+      return list;
+    }
+
+    const valueFieldName = 'subscriptionId';
+    const textFieldName = 'displayName';
+    for (let i = 0; i < result.data.value.length; i++) {
+      if (!find(list, ['value', get(result.data.value[i], valueFieldName)])) {
+        list.push({
+          text: `${get(result.data.value[i], textFieldName)}`,
```

Comment:
not sure if the value can be an object, in that case text would be `[object Object]`. Is that expected?

Reply:
I copied it over as is from `azure-monitor-datasource`.

As mentioned above, we need to overhaul all these methods at once.

I created a separated work item clean this up:
#34860